man/zram-generator.md

```diff
@@ -0,0 +1,57 @@
+=================================================================
+
+## SYNOPSIS
+
+`/lib/systemd/system-generators/zram-generator` `TARGET_DIR` [*2RGET_DIR* *3RGET_DIR*]<br />
+`/lib/systemd/system-generators/zram-generator` --setup-device `DEVICE`<br />
+`/lib/systemd/system-generators/zram-generator` --reset-device `DEVICE`
+
+## DESCRIPTION
+
+`zram-generator` is a generator that creates zram devices and systemd units to configure, format, and enable swap on them.
+
+The generator will be invoked by systemd early at boot, the generator will then:
+
+  1. read configuration files from *{/etc,/lib}/systemd/zram-generator.conf[.d]* (see zram-generator.conf(5) for details);
+  2. if systemd-detect-virt(1) exists and returns non-*0*, exit, since continuing would mean changing the host system;
+  3. generate systemd.swap(5) units into `TARGET_DIR` for all configured devices and symlink them into *swap.target.wants/*;
+  4. if any devices are enabled, and the zram module isn't loaded (*/sys/class/zram-control* doesn't exist), invoke *modprobe(8) zram*;
+  5. if the requested devices aren't present, read */sys/class/zram-control/hot_add* until the highest device's number is read,
+
+If the `ZRAM_GENERATOR_ROOT` environment variable is present, the return value, if any, of systemd-detect-virt(1) is ignored, and steps `4` and `5` are skipped;
+for the ramifications of `ZRAM_GENERATOR_ROOT` on config handling, see zram-generator.conf(5).
+
+
+The generated *dev-zramN.swap* units depend on `zram-create@zramN.service`, which on start invokes the generator with *--setup-device zramN*, which will then:
+
+  1. read configuration files from *{/etc,/lib}/systemd/zram-generator.conf[.d]* (see zram-generator.conf(5) for details);
+  2. set the desired compression algorithm in */sys/block/zramN/comp_algorithm*, if any;
+     if the write fails with `EINVAL` the current kernel doesn't understand the specified algorithm; a warning is issued, but execution continues as normal,
+     if the write fails for any other reason, setup is halted;
+  3. set the desired blockdev size in */sys/block/zramN/disksize*, if the write fails, setup is halted;
+  4. run *systemd-makefs(8) swap /dev/zramN*
+
+And on stop with *--reset-device zramN*, which will write a positive integer to */sys/block/zramN/reset*, allowing the device to be reused.
+
+
+`zram-generator` implements systemd.generator(7).
+
+## AUTHOR
+
+Written by Zbigniew Jędrzejewski-Szmek &lt;<zbyszek@in.waw.pl>&gt;,
+           Igor Raits &lt;<i.gnatenko.brain@gmail.com>&gt;,
+       and наб &lt;<nabijaczleweli@gmail.com>&gt;
+
+## REPORTING BUGS
+
+&lt;<https://github.com/systemd/zram-generator/issues>&gt;
+
+## SEE ALSO
+
+zram-generator.conf(5), systemd.generator(7)
+
+&lt;<https://github.com/systemd/zram-generator>&gt;
+
+Linux documentation of zram: &lt;<https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/admin-guide/blockdev/zram.rst?h=v5.7>&gt;<br />
```

Comment:
https://www.kernel.org/doc/html/latest/admin-guide/blockdev/zram.html

Reply:
Oh, cool. I've made it a versioned link though, since these things can and do move (it's in Documentation/blockdev/zram.rst on the 4.19 buster kernel, but moved to admin-guide/ sometime since), and usually nobody notices. Indeed, I recently fixed a whole host of these in the util-linux manpages which nobody noticed since 2016.

Reply:
I'd prefer to keep the "dynamic" link. There's nothing specific about kernel 5.7, and we want to always refer to the latest version of the kernel documentation. If the link becomes stale, it'll be easy to fix. (Though, I hope that the kernel learns to do redirects for link stability so this won't be necessary.)

Reply:
Done